### PR TITLE
BoundedСontext, Engine no more.

### PR DESCRIPTION
**Pull request by alexander-yevsyukov**

### CODE REVIEW

File: core-java-sample/src/main/java/org/spine3/sample/Application.java [hunk 1]
```diff
@@ -126,10 +126,10 @@ public void tearDown() {
         storageFactory.tearDown();
 
         // Unregister event handlers
-        engine.getEventBus().unregister(eventLogger);
+        boundedContext.getEventBus().unregister(eventLogger);
 
         // Stop the engine
-        engine.stop();
+        boundedContext.stop();
```
Comment (AlexanderLitus) [inline core-java-sample/src/main/java/org/spine3/sample/Application.java]: I think we can rename `stop` to `tearDown` or something.

File: core-java-sample/src/main/java/org/spine3/sample/Application.java [hunk 2]
```diff
@@ -126,10 +126,10 @@ public void tearDown() {
         storageFactory.tearDown();
 
         // Unregister event handlers
-        engine.getEventBus().unregister(eventLogger);
+        boundedContext.getEventBus().unregister(eventLogger);
 
         // Stop the engine
```
Comment (AlexanderLitus) [inline core-java-sample/src/main/java/org/spine3/sample/Application.java]: This comment is deprecated.

File: core-java/src/main/java/org/spine3/server/Repository.java [hunk 3]
```diff
@@ -39,10 +39,10 @@
      *
      * <p>The type of the storage depends on and should be checked by the implementations.
      *
-     * <p>This method should be normally called once during registration of the repository with {@link Engine}.
+     * <p>This method should be normally called once during registration of the repository with {@link BoundedContext}.
      * An attempt to call this method twice with different parameters will cause {@link IllegalStateException}.
      *
-     * <p>{@link Engine} will call this method with {@code null} argument to request performing all necessary
+     * <p>{@link BoundedContext} will call this method with {@code null} argument to request performing all necessary
      * clean-up operations before the the engine stops.
```
Comment (AlexanderLitus) [inline core-java/src/main/java/org/spine3/server/Repository.java]: `...before the the engine stops.` is deprecated.

**Comment by alexander-yevsyukov** [comment]
@AlexanderLitus, PTAL.

**Comment by AlexanderLitus** [comment]
LGTM.

**Pull request merged**